File: day7/.js
/** @param {number} height - Height of the tree
 *  @param {string} ornament - Character to use as ornament
 *  @param {number} frequency - How often ornaments appear
 *  @returns {string} The decorated tree
 */
function drawTree(height, ornament, frequency) {
  let string = "";
  let pos = 1;

  for (let i = 0; i < height; i++) {
    let line = "";
    for (let j = 0; j < 2 * i + 1; j++) {
      line += pos % frequency == 0 ? ornament : "*";
      pos++;
    }
    string += " ".repeat(height - i - 1) + line + "\n";
  }
  string += " ".repeat(height - 1) + "#";
  return string;
}

File: day07/.py
def draw_tree(height, ornament, frequency):
  tree_drawing = ""
  pos = 1

  for i in range(height):
    line = ""
    for j in range(2*i+1):
      line+= ornament if pos % frequency == 0 else "*"
      pos += 1
    tree_drawing += " "*(height-i-1) + line + "\n"
  tree_drawing += " "*(height-1) + "#"
  return tree_drawing
